### PR TITLE
Add overlay windows handling

### PR DESCRIPTION
**Pull request by faenil**

### CODE REVIEW

**Comment by locusf** [comment]
Tested as ok on N9.

File: src/qml/compositor.qml [hunk 1]
```diff
@@ -68,13 +68,27 @@ Compositor {
...
+        onActiveFocusItemChanged: {
+            // Search for the layer of the focus item
+            var focusedLayer = root.activeFocusItem
+            while (focusedLayer && focusedLayer.parent !== layersParent)
+                focusedLayer = focusedLayer.parent
+
+            // reparent the overlay to the found layer
+            overlayLayer.parent = focusedLayer ? focusedLayer : overlayLayer.parent
+        }
+    }
```
Comment (special) [inline src/qml/compositor.qml:82]: Hmm, is the reparenting actually necessary, since the overlayLayer has a higher Z than any of the focus item layers?

Comment (faenil) [inline src/qml/compositor.qml:82]: Honestly, I trusted you on this :)

Yes, doesn't make much sense to me, but iirc it didn't work correctly without it

**Comment by special** [comment]
Otherwise LGTM. Hopefully alpha for wayland items gets fixed at some point so the shader becomes unnecessary.

**Comment by faenil** [comment]
About reparenting without setting Z:

I did some tests, reparenting without setting overlayLayer's higher Z doesn't work, in fact in that case if you open app A and show keyboard and then open app B, the keyboard will show in app A but not in app B, as the reparenting code just sets the parent once in that case (as the focusedLayer will still be appLayer) but app B will be drawn on top of the keyboard as its window has been created after keyboard's one, afaiu.

About setting Z without reparenting:
this seems to work ok...though we have the issue of maliit appearing on windows where it's not needed. This seems to happen in the case described above as well...
Basically, you open keyboard in app A, then send A to background, and keyboard closes. Then you open fingerterm or dialer (for example) and they keyboard appears fully open again, and starts animating to bottom to reach closed mode (and it also happens that it gets stuck during the animation!).

I think fixing that issue could be part of this same PR, so I'll wait for your opinion on this before merging, I only spent very short time on this as I've been busy

**Comment by locusf** [comment]
Merging as per LGTM.
